db/change_cache.go

```diff
@@ -317,68 +315,8 @@ func (c *changeCache) CleanSkippedSequenceQueue(ctx context.Context) error {
-		} else {
-			skippedSeqBatch = oldSkippedSequences[0:]
-			oldSkippedSequences = nil
-		}
-
-		base.InfofCtx(ctx, base.KeyCache, "Issuing skipped sequence clean query for %d sequences, %d remain pending (db:%s).", len(skippedSeqBatch), len(oldSkippedSequences), base.MD(c.collection.Name()))
-		// Note: The view query is only going to hit for active revisions - sequences associated with inactive revisions
-		//       aren't indexed by the channel view.  This means we can potentially miss channel removals:
-		//       when an older revision is missed by the TAP feed, and a channel is removed in that revision,
-		//       the doc won't be flagged as removed from that channel in the in-memory channel cache.
-		entries, err := c.collection.getChangesForSequences(ctx, skippedSeqBatch)
-		if err != nil {
-			base.WarnfCtx(ctx, "Error retrieving sequences via query during skipped sequence clean - #%d sequences treated as not found: %v", len(skippedSeqBatch), err)
-			continue
-		}
-
-		// Process found entries.  Add to foundEntries for subsequent cache processing, foundMap for pendingRemoval calculation below.
-		foundMap := make(map[uint64]struct{}, len(entries))
-		for _, foundEntry := range entries {
-			foundMap[foundEntry.Sequence] = struct{}{}
-			foundEntries = append(foundEntries, foundEntry)
-		}
-
-		// Add queried sequences not in the resultset to pendingRemovals
-		for _, skippedSeq := range skippedSeqBatch {
-			if _, ok := foundMap[skippedSeq]; !ok {
-				base.WarnfCtx(ctx, "Skipped Sequence %d didn't show up in MaxChannelLogMissingWaitTime, and isn't available from a * channel query.  If it's a valid sequence, it won't be replicated until Sync Gateway is restarted.", skippedSeq)
-				pendingRemovals = append(pendingRemovals, skippedSeq)
-			}
-		}
-	}
-
-	// Issue processEntry for found entries.  Standard processEntry handling will remove these sequences from the skipped seq queue.
-	changedChannelsCombined := channels.Set{}
-	for _, entry := range foundEntries {
-		entry.Skipped = true
-		// Need to populate the actual channels for this entry - the entry returned from the * channel
-		// view will only have the * channel
-		doc, err := c.collection.GetDocument(ctx, entry.DocID, DocUnmarshalNoHistory)
-		if err != nil {
-			base.WarnfCtx(ctx, "Unable to retrieve doc when processing skipped document %q: abandoning sequence %d", base.UD(entry.DocID), entry.Sequence)
-			continue
-		}
-		entry.Channels = doc.Channels
-
-		changedChannels := c.processEntry(entry)
-		changedChannelsCombined = changedChannelsCombined.Update(changedChannels)
-	}
-
-	// Since the calls to processEntry() above may unblock pending sequences, if there were any changed channels we need
-	// to notify any change listeners that are working changes feeds for these channels
-	if c.notifyChange != nil && len(changedChannelsCombined) > 0 {
-		c.notifyChange(changedChannelsCombined)
 	}
+	oldSkippedSequences = nil
```

Comment:
We need to remove the condition on line 316, I believe, in order to properly populate pendingRemovals.